File: src/main/kotlin/xyz/lappe/timetracker/model/TimerInstance.kt
package xyz.lappe.timetracker.model

import java.time.Instant
import java.time.temporal.ChronoUnit
import javax.persistence.Entity
import javax.persistence.GeneratedValue
import javax.persistence.Id

@Entity
public class TimerInstance(@Id @GeneratedValue var id: Long? = null, var name: String, var startTime: Instant, var endTime: Instant? = null) {
    constructor() : this(name = "", startTime = Instant.MIN) {
    }


    fun duration() : Long = startTime.until(endTime, ChronoUnit.SECONDS)
}

File: frontend/src/model/TimerInstance.ts
import moment, {Moment} from 'moment';

export interface TimerInstance {
  id?: number;
  name: string;
  startTime: Moment;
  endTime?: Moment;
}

export class TimerImpl implements TimerInstance {
  id?: number;
  name: string;
  startTime: Moment;
  endTime?: Moment;


  constructor(id: number | undefined, name: string, startTime: Moment, endTime: Moment | undefined) {
    this.id = id;
    this.name = name;
    this.startTime = startTime;
    this.endTime = endTime;
  }

  static parse(timer: TimerInstance): TimerImpl {
    return new TimerImpl(
      timer.id,
      timer.name,
      moment(timer.startTime),
      timer.endTime
        ? moment(timer.endTime)
        : undefined)
  }
}
